extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/ClassRestClientContext.java

```diff
@@ -125,4 +129,43 @@ protected Supplier<FieldDescriptor> getLazyJavaMethodGenericParametersField(int
             return javaMethodGenericParametersField;
         };
     }
+
+    /**
+     * Generates "Class.forName(beanClass)" to generate the parameter descriptors. This method will only be created if and only
+     * if the supplier is used in order to not have a penalty performance.
+     */
+    protected Supplier<FieldDescriptor> getLazyBeanParameterDescriptors(String beanClass) {
+        return () -> {
+            FieldDescriptor field = beanTypesParameterDescriptorsStaticFields.get(beanClass);
+            if (field != null) {
+                return field;
+            }
+
+            ResultHandle clazz = loadClass(beanClass);
+
+            ResultHandle mapWithAnnotationsHandle = clinit.newInstance(MethodDescriptor.ofConstructor(
+                    ParameterDescriptorFromClassSupplier.class, Class.class),
+                    clazz);
+            field = FieldDescriptor.of(classCreator.getClassName(), "beanParamDescriptors" + beanParamIndex, Supplier.class);
+            classCreator.getFieldCreator(field).setModifiers(Modifier.FINAL | Modifier.STATIC); // needs to be package-private because it's used by subresources
+            clinit.writeStaticField(field, mapWithAnnotationsHandle);
+
+            beanTypesParameterDescriptorsStaticFields.put(beanClass, field);
+
+            beanParamIndex++;
+
+            return field;
+        };
+    }
+
+    private ResultHandle loadClass(String className) {
```

Comment:
This method is to avoid loading the same class only once, so we avoid a performance penalty here.